storage/grpc_client.go

```diff
@@ -3021,6 +3037,29 @@ func (w *gRPCWriter) read() (int, bool, error) {
 	return recvd, done, err
 }
 
+// flush flushes the current buffer regardless of whether it is full or not.
+// It's the implementation for Writer.Flush.
+func (w *gRPCWriter) flush() (int64, error) {
+	log.Printf("calling flush")
+	if !w.append {
+		return 0, errors.New("Flush is supported only if Writer.Append is set to true")
+	}
+
+	// Close PipeWriter to trigger EOF on read side of the stream.
+	w.flushInProgress = true
```

Comment:
I think this would be nicer as a bool chan - we only need to consume it in one place, right?
Also, I don't think it's getting set back to false anywhere (which wouldn't be necessary if it were a chan)?

Reply:
This should be fixed now?